test-case/multiple-pause-resume.sh

```diff
@@ -0,0 +1,212 @@
+OPT_PARM_lst['i']=1         OPT_VALUE_lst['i']='100'
+
+OPT_OPT_lst['a']='max'      OPT_DESC_lst['a']='random range max value, unit is ms'
+OPT_PARM_lst['a']=1         OPT_VALUE_lst['a']='200'
+
+OPT_OPT_lst['s']='sof-logger'   OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0             OPT_VALUE_lst['s']=1
+
+func_opt_parse_option $*
+
+repeat_count=${OPT_VALUE_lst['r']}
+loop_count=${OPT_VALUE_lst['l']}
+# configure random value range
+rnd_min=${OPT_VALUE_lst['i']}
+rnd_max=${OPT_VALUE_lst['a']}
+rnd_range=$[ $rnd_max - $rnd_min ]
+[[ $rnd_range -le 0 ]] && dlogw "Error random range scope [ min:$rnd_min - max:$rnd_max ]" && exit 2
+
+tplg=${OPT_VALUE_lst['t']}
+func_pipeline_export $tplg
+# get the min value of TPLG:'pipeline count' with Case:'pipeline count'
+[[ $PIPELINE_COUNT -gt ${OPT_VALUE_lst['c']} ]] && max_count=${OPT_VALUE_lst['c']} || max_count=$PIPELINE_COUNT
+[[ $max_count -eq 1 ]] && dlogw "pipeline count is 1, don't need to run this case" && exit 2
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+func_lib_setup_kernel_last_line
+
+declare -a pipeline_idx_lst
+declare -a cmd_idx_lst
+declare -a file_idx_lst
+
+# merge all pipeline to the 1 group
+for i in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+do
+    pipeline_idx_lst=(${pipeline_idx_lst[*]} $i)
+    type=$(func_pipeline_parse_value $i type)
+    if [ "$type" == "playback" ];then
+        cmd_idx_lst=(${cmd_idx_lst[*]} "aplay")
+        file_idx_lst=(${file_idx_lst[*]} "/dev/zero")
+    elif [ "$type" == "capture" ];then
+        cmd_idx_lst=(${cmd_idx_lst[*]} "arecord")
+        file_idx_lst=(${file_idx_lst[*]} "/dev/null")
+    elif [ "$type" == "both" ];then
+        cmd_idx_lst=(${cmd_idx_lst[*]} "aplay")
+        file_idx_lst=(${file_idx_lst[*]} "/dev/zero")
+        # both include playback & capture, so duplicate it
+        pipeline_idx_lst=(${pipeline_idx_lst[*]} $i)
+        cmd_idx_lst=(${cmd_idx_lst[*]} "arecord")
+        file_idx_lst=(${file_idx_lst[*]} "/dev/null")
+    else
+        dloge "Unknow pipeline type: $type"
+        exit 1
+    fi
+done
+pipeline_list_count=$(expr ${#pipeline_idx_lst[*]} - 1)
+# x here as the flag for pipeline split
+tmp_lst_str=$(echo ${pipeline_idx_lst[*]}|sed "s/ /x,/g")"x"
+for i in $(seq 1 $max_count)
+do
+    tmp_str="$tmp_str{$tmp_lst_str}"
+done
+unset tmp_lst_str
+# skill of bash for permutation & Combinatorics 
+# echo {0x,1x,2x}{0x,1x,2x}
+# 0x0x 0x1x 0x2x 1x0x 1x1x 1x2x 2x0x 2x1x 2x2x
+# current this case just need combine so remove permutation
+for i in $(echo $(eval echo $tmp_str))
+do
+    # sort the order, 1x0x -> 0x1x; 2x1x -> 1x2x
+    pipeline_combine="$(echo $i |tr 'x' '\n' |sort|tr '\n' 'x'|sed 's/^x//g;s/x$//g')"
+    # this combine already saved
+    [[ "$(echo $pipeline_lst_str|grep "$pipeline_combine")" ]] && continue
+    pipeline_lst_str="$pipeline_lst_str $pipeline_combine"
+done
```

Comment:
the final value should be 0x0x0x, 0x0x1 0x1x0 0x1x1 1x0x0 1x0x1 ... ? Is this what you wanted?

Reply:
`# sort the order, 1x0x -> 0x1x; 2x1x -> 1x2x`
so here `pipeline_combine` is order by it's element
So `0x0x 0x1x 0x2x 1x0x 1x1x 1x2x 2x0x 2x1x 2x2x`
will become `0x0x 0x1x 0x2x 0x1x 1x1x 1x2x 0x2x 1x2x 2x2x`
Now you can find out whether this string is used